sdks/python/apache_beam/transforms/util.py

```diff
@@ -825,7 +827,60 @@ def expand(self, pcoll):
               key_value[1]))
       return (
           sharded_pcoll
-          | GroupIntoBatches(self.batch_size, self.max_buffering_duration_secs))
+          | GroupIntoBatches(
+              self.params.batch_size, self.params.max_buffering_duration_secs))
+
+    def to_runner_api_parameter(
+        self,
+        unused_context  # type: PipelineContext
+    ):  # type: (...) -> Tuple[str, beam_runner_api_pb2.GroupIntoBatchesPayload]
+      return (
+          common_urns.composites.GROUP_INTO_BATCHES_WITH_SHARDED_KEY.urn,
+          self.params.get_payload())
+
+    @staticmethod
+    @PTransform.register_urn(
+        common_urns.composites.GROUP_INTO_BATCHES_WITH_SHARDED_KEY.urn,
+        beam_runner_api_pb2.GroupIntoBatchesPayload)
+    def from_runner_api_parameter(unused_ptransform, proto, unused_context):
+      return GroupIntoBatches.WithShardedKey(
+          *GroupIntoBatchesParams.parse_payload(proto))
+
+
+class GroupIntoBatchesParams:
```

Comment:
Please add some pydoc to describe this class. Other than this, this PR is ready to go.

Reply:
Done. I also changed the class name to `_GroupIntoBatchesParams`; it's not intended to be "public" and referenced outside of this file.